### DISCUSSION TITLE
/image should support links

### DISCUSSION THREAD

**Discussion by ChuckJonas** [outline/outline#2994] [General]
Currently when you add a via `/image` it opens the system file prompt.

It would be nice if there was an option to instead paste in a URL.  I know you can accomplish this with markdown `![]()` syntax, but many of our users are not that advanced.

**Comment by ChuckJonas** [comment]
Also noticed today, if you add an image with markdown, if you click "replace", you only get the option to upload a replacement.

**Comment by roovyshapiro** [comment]
Agreed! I frequently use imgur to host screenshots and my only options are to either paste the link or upload the actual image file. I would love to be able to post an image URL and have it fully displayed on the page. Youtube URLs do this natively in outline so it would be great for images too.

**Comment by tommoor** [comment]
Definitely want to allow this but it does mean the creation of an intermediate dialog to let you choose between "From desktop" / "From url" (and potentially other future options like "From Google Drive") when adding an image form the block menu.

**Reply by roovyshapiro** [reply]
An intermediate dialog is perfectly fine.
I just tested how Notion does this as a reference. 
When pasting an image link, an auto dialog opens that allows you embed it.
When putting in an image, a dialog opens that allows you to choose to upload a file or paste a URL.
![](https://i.imgur.com/lHqi56W.gif)
Both options were very comfortable. When you type "/image" in outline it immediately opens up uploading an image, but I don't feel that much is lost with the extra dialog, and the ability to embed images would be worth it.

**Reply by JasonOui1** [reply]
Could we potentially have another shortcut such as `/imageurl` to embed an image from a URL?

**Reply by roovyshapiro** [reply]
I've been consistently using `![](image link here) `and its been working great for images and gifs. I had to educate my team how to do it, but honestly its been fine.
`![](https://i.imgur.com/JPX4FwT.png)`
![](https://i.imgur.com/JPX4FwT.png)
